libcontainer/nsenter/nsexec.c

```diff
@@ -164,7 +164,8 @@ static void write_log(const char *level, const char *format, ...)
 		goto out;
 	}
 
-	write(logfd, json, ret);
+	if (write(logfd, json, ret) != ret)
+		goto out;
```

Comment:
this is useless...

A way to suppress a warning is `(void)write(...)`

Reply:
Simply casting to (void) won't work. See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=25509 for detailed info.

Alternatives below if we really don't want to check the return value, which may however lose kind of readability:
1) (void)!write(...)
2) if (write(...));

What's the preference? Thanks! @kolyshkin

Reply:
Oh my...

Frankly, I don't know. What's the best way from the readability point of view?

Perhaps something like

```c
if (write(logfd, json, ret) < 0)
	; // nothing we can do here
```

Reply:
Updated. PTAL, thanks! @kolyshkin

Reply:
I guess that `if` stmt may break another linter.

Isn't there an gcc `__attribute__` to ignore the warning? (I'm googling around...)

Reply:
`#pragma GCC diagnostic ignored "-Wunused-result"` may work (untested)

Reply:
@AkihiroSuda Thanks for the comment!

emm yes, I agree this may break another linter. But `#pragma GCC diagnostic ignored "-Wunused-result"` won't work for older gcc versions before `Diagnostic Pragmas` was introduced.

Do you think adding braces to avoid linters checking empty bodies for if statements work for you?
```
if (write(logfd, json, ret) < 0) {
	; // nothing we can do here
}
```

Reply:
Updated. PTAL thanks! @kolyshkin @AkihiroSuda

Reply:
Even gcc4 seems supporting that pragma https://stackoverflow.com/questions/63229590/enabling-long-long-in-c89-at-gcc-3-2-4-4-and-5-4

So we can safely use the pragma

Reply:
@AkihiroSuda Thanks for checking this!

Updated using pragmas, PTAL. Thanks!